### DISCUSSION TITLE
Can I add a new Log-Level?

### DISCUSSION THREAD

**Discussion by mobergmann** [gabime/spdlog#2703] [Q&A] [answered]
I just discovered this library and I want to integrate it into oure project. I asked myself if it possible to add a new log level, which I then can use like the default log-levels (like so `spdlog::my_loglevel("Hello World")`)? This is not a requirement, but it would be a nice to have.

When it is possible, I would also have to adjust the hierarchic of log levels, maybe add a custom color and probably add many more things, which I currently cannot imagine.

**Comment by tt4g** [comment, marked as answer]
If you want to add an API to the `spdlog` namespace, you have to edit the source files.

Any log level that is printed in the log message can be overridden by the macro.

https://github.com/gabime/spdlog/blob/ad0e89cbfb4d0c1ce4d097e134eb7be67baebb36/include/spdlog/tweakme.h#L103-L114

I'm not exactly sure what "adjust the hierarchic of log levels" means, but if you want to change the behavior when comparing log levels, change the number assigned to the log level.

https://github.com/gabime/spdlog/blob/ad0e89cbfb4d0c1ce4d097e134eb7be67baebb36/include/spdlog/common.h#L198-L222

See #2279 how to set up custom colors.

See also: #1794, #834

**Reply by mobergmann** [reply]
I mainly wanted to know if there is a convenient way to add a new log level. But editing the source files is not convenient. So I take this as a no, it is not possible.  Thanks for the fast response :+1:
